### PR TITLE
Adding deploy script for TeleporterRegistry

### PR DESCRIPTION
**Pull request by minghinmatthewlam**
## Why this should be merged
Deploy script for TeleporterRegistry

## How this works
Updates `release.yaml` to include registry byte code
Adds a script to deploy registry
Updates README

## How this was tested
deploy to dispatch and c-chain

## How is this documented
README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,33 +178,39 @@ For more information on the registry and how to integrate with Teleporter dApps,
...
 
 ## Deploy TeleporterRegistry to a Subnet
 
-There should only be one canonical `TeleporterRegistry` deployed for each chain, but if one does not exist, it is recommended to deploy the registry so Teleporter dApps can always use the most recent Teleporter version available. Since the registry does not need to be deployed to the same address on every chain, it does not need a Nick's method deployment, and can be deployed using `forge create` from the root of the repo:
+There should only be one canonical `TeleporterRegistry` deployed for each chain, but if one does not exist, it is recommended to deploy the registry so Teleporter dApps can always use the most recent Teleporter version available. The registry does not need to be deployed to the same address on every chain, and does not need a Nick's method transaction to deploy
```
Comment (michaelkaplan13) [inline README.md:199]: Incomplete sentence here.

File: scripts/deploy_registry.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+function printUsage() {
+    cat << EOF
+Arguments:
+    --version <version>              Specify the release version to deploy.
+    --rpc-url <url>                  Specify the rpc url of the node to use
+    --private-key <private_key>      Private key of account to deploy TeleporterRegistry
+Options:
+    --constructor-args '[(<teleporter_version>,<teleporter_contract_address)...]'
```
Comment (michaelkaplan13) [inline scripts/deploy_registry.sh:37]: Missing close > here

File: scripts/deploy_registry.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+if [[ $teleporter_version == "" || $rpc_url == "" || $user_private_key == "" ]]; then
+    echo "Invalid usage. Missing required command line arguments."
+    printHelp && exit 1
+fi
+
+teleporter_registry_bytecode=$(curl -sL https://github.com/ava-labs/teleporter/releases/download/$teleporter_version/TeleporterRegistry_Bytecode_$teleporter_version.txt)
+teleporter_contract_address=$(curl -sL https://github.com/ava-labs/teleporter/releases/download/$teleporter_version/TeleporterMessenger_Contract_Address_$teleporter_version.txt)
+if [ "$teleporter_registry_bytecode" == "Not Found" ]; then
+  echo "Error: TeleporterRegistry $teleporter_version byte code not found."
+  exit 1
```
Comment (geoff-vball) [inline scripts/deploy_registry.sh:85]: nit: these are spaced less than the other indents

File: scripts/deploy_teleporter.sh [hunk 4]
```diff
@@ -86,6 +90,10 @@ teleporter_deployer_address=$(curl -sL https://github.com/ava-labs/teleporter/re
 echo "TeleporterMessenger $teleporter_version deployer address: $teleporter_deployer_address"
 teleporter_deploy_tx=$(curl -sL https://github.com/ava-labs/teleporter/releases/download/$teleporter_version/TeleporterMessenger_Deployment_Transaction_$teleporter_version.txt)
 teleporter_messenger_bytecode=$(curl -sL https://github.com/ava-labs/teleporter/releases/download/$teleporter_version/TeleporterMessenger_Bytecode_$teleporter_version.txt)
+if [ "$teleporter_contract_address" == "Not Found" ]; then
+  echo "Error: TeleporterMessenger $teleporter_version contract address not found."
+  exit 1
```
Comment (geoff-vball) [inline scripts/deploy_teleporter.sh:95]: nit: indentation

File: README.md [hunk 5]
```diff
@@ -178,33 +178,37 @@ For more information on the registry and how to integrate with Teleporter dApps,
...
 
 ## Deploy TeleporterRegistry to a Subnet
 
-There should only be one canonical `TeleporterRegistry` deployed for each chain, but if one does not exist, it is recommended to deploy the registry so Teleporter dApps can always use the most recent Teleporter version available. Since the registry does not need to be deployed to the same address on every chain, it does not need a Nick's method deployment, and can be deployed using `forge create` from the root of the repo:
+There should only be one canonical `TeleporterRegistry` deployed for each chain, but if one does not exist, it is recommended to deploy the registry so Teleporter dApps can always use the most recent Teleporter version available. The registry does not need to be deployed to the same address on every chain, therefore does not need a Nick's method transaction. To deploy, run the following command from root:
```
Comment (cam-schultz) [inline README.md:199]:
```suggestion
There should only be one canonical `TeleporterRegistry` deployed for each chain, but if one does not exist, it is recommended to deploy the registry so Teleporter dApps can always use the most recent Teleporter version available. The registry does not need to be deployed to the same address on every chain, therefore does not need a Nick's method transaction. To deploy, run the following command from the repository root:
```

File: README.md [hunk 6]
```diff
@@ -178,33 +178,37 @@ For more information on the registry and how to integrate with Teleporter dApps,
...
+./scripts/deploy_registry.sh --version <version> --rpc-url <url> --private-key <private_key> [OPTIONS]
 ```
 
-- `$user_private_key`: private key of the user deploying the contract.
-- `$subnet_rpc_url`: RPC URL of the subnet where the contract will be deployed.
-- `$teleporter_version`: version number of the `TeleporterMessenger` contract being registered on this blockchain. For example, if it's the first `TeleporterMessenger` contract being registered, the version number would be 1.
-- `$teleporter_contract_address`: address of the `TeleporterMessenger` contract being registered on this blockchain.
+Required arguments:
+
+- `--version <version>` Specify the release version to deploy. These will all be of the form `v1.X.0`. Each Teleporter version can only send and receive messages from the **same** Teleporter version on another chain. You can see a list of released versions at https://github.com/ava-labs/teleporter/releases.
```
Comment (cam-schultz) [inline README.md:207]: I think we should clarify that TeleporterRegistries on different chains do not need to be the same version.

File: README.md (hunk 5 above)
Comment (michaelkaplan13) [reply README.md:199]: Or "from the root of the repository" 👍

File: README.md [hunk 7]
```diff
@@ -178,33 +178,37 @@ For more information on the registry and how to integrate with Teleporter dApps,
...
+- `--rpc-url <url>` Specify the rpc url of the node to use.
+- `--fund-deployer <private_key>` Funds the deployer address with the account held by `<private_key>`
```
Comment (michaelkaplan13) [inline README.md:209]: I think this should be --private-key

File: README.md [hunk 8]
```diff
@@ -178,33 +178,37 @@ For more information on the registry and how to integrate with Teleporter dApps,
...
 
-- `--version <version>` Required. Specify the release version to deploy. These will all be of the form `v1.X.0`. Each Teleporter version can only send and receive messages from the **same** Teleporter version on another chain. You can see a list of released versions at https://github.com/ava-labs/teleporter/releases.
-- `--rpc-url <url>` Required. Specify the rpc url of the node to use.
-- `--fund-deployer <private_key>` Optional. Funds the deployer address with the account held by `<private_key>`
+- `--version <version>` Specify the release version to deploy. These will all be of the form `v1.X.0`. Each Teleporter version can only send and receive messages from the **same** Teleporter version on another chain. You can see a list of released versions at https://github.com/ava-labs/teleporter/releases.
+- `--rpc-url <url>` Specify the rpc url of the node to use.
+
+Options:
+
+- `--fund-deployer <private_key>` Funds the deployer address with the account held by `<private_key>`
```
Comment (michaelkaplan13) [inline README.md:191]: I think this should be --private-key also. My bad I forgot to update it before.

**Review by cam-schultz — APPROVED** [review]

File: README.md [hunk 9]
```diff
@@ -178,33 +178,37 @@ For more information on the registry and how to integrate with Teleporter dApps,
...
 
 ## Deploy TeleporterRegistry to a Subnet
 
-There should only be one canonical `TeleporterRegistry` deployed for each chain, but if one does not exist, it is recommended to deploy the registry so Teleporter dApps can always use the most recent Teleporter version available. Since the registry does not need to be deployed to the same address on every chain, it does not need a Nick's method deployment, and can be deployed using `forge create` from the root of the repo:
+There should only be one canonical `TeleporterRegistry` deployed for each chain, but if one does not exist, it is recommended to deploy the registry so Teleporter dApps can always use the most recent Teleporter version available. The registry does not need to be deployed to the same address on every chain, therefore does not need a Nick's method transaction. To deploy, run the following command from the root of the repository:
```
Comment (michaelkaplan13) [inline README.md:199]:
```suggestion
There should only be one canonical `TeleporterRegistry` deployed for each chain, but if one does not exist, it is recommended to deploy the registry so Teleporter dApps can always use the most recent Teleporter version available. The registry does not need to be deployed to the same address on every chain, and therefore does not need a Nick's method transaction. To deploy, run the following command from the root of the repository:
```

**Review by cam-schultz — APPROVED** [review]

**Review by michaelkaplan13 — APPROVED** [review]

**Review by geoff-vball — APPROVED** [review]

**Pull request merged**